procedures/diagnostics.sql

```diff
@@ -0,0 +1,1168 @@
+        '''last_query_cost'', ',
+        '''last_query_partial_plans'', ',
+        '''qcache_total_blocks'', ',
+        '''slave_last_heartbeat'', ',
+        '''ssl_ctx_verify_depth'', ',
+        '''ssl_ctx_verify_mode'', ',
+        '''ssl_session_cache_size'', ',
+        '''ssl_verify_depth'', ',
+        '''ssl_verify_mode'', ',
+        '''ssl_version'', ',
+        '''buffer_flush_lsn_avg_rate'', ',
+        '''buffer_flush_pct_for_dirty'', ',
+        '''buffer_flush_pct_for_lsn'', ',
+        '''buffer_pool_pages_total'', ',
+        '''lock_row_lock_time_avg'', ',
+        '''lock_row_lock_time_max'', ',
+        '''innodb_page_size''',
+    ')');
+
+    -- Do not track the current thread - no reason to clutter the output
+    SELECT INSTRUMENTED INTO v_this_thread_enabled FROM performance_schema.threads WHERE PROCESSLIST_ID = CONNECTION_ID();
+    CALL sys.ps_setup_disable_thread(CONNECTION_ID());
+
+    -- Set configuration options
+    IF (@sys.allow_i_s_tables IS NULL) THEN
+        SET @sys.allow_i_s_tables          = sys.sys_get_config('allow_i_s_tables'         , 'OFF');
+    END IF;
+    IF (@sys.diagnostics.include_raw IS NULL) THEN
+        SET @sys.diagnostics.include_raw   = sys.sys_get_config('diagnostics.include_raw'  , 'OFF');
+    END IF;
+    IF (@sys.debug IS NULL) THEN
+        SET @sys.debug                     = sys.sys_get_config('debug'                    , 'OFF');
+    END IF;
+    IF (@sys.statement_truncate_len IS NULL) THEN
+        SET @sys.statement_truncate_len    = sys.sys_get_config('statement_truncate_len'   , '64' );
+    END IF;
+
+    -- Temorary table are used - disable sql_log_bin if necessary to prevent them replicating
+    SET @log_bin := @@sql_log_bin;
+    IF (@log_bin = 1) THEN
+        SET sql_log_bin = 0;
+    END IF;
+
+    IF (in_auto_config <> 'current') THEN
+        IF (@sys.debug = 'ON') THEN
+            SELECT CONCAT('Updating Performance Schema configuration to ', in_auto_config) AS 'Debug';
+        END IF;
+        CALL sys.ps_setup_save(0);
+
+        IF (in_auto_config = 'medium') THEN
+            -- Enable all consumers except %history and %history_long
+            UPDATE performance_schema.setup_consumers
+                SET ENABLED = 'YES'
+            WHERE NAME NOT LIKE '%\_history%';
+
+            -- Enable all instruments except wait/synch/%
+            UPDATE performance_schema.setup_instruments
+                SET ENABLED = 'YES',
+                    TIMED   = 'YES'
+            WHERE NAME NOT LIKE 'wait/synch/%';
+
+            -- Enable all threads
+            UPDATE performance_schema.threads
+                SET INSTRUMENTED = 'YES';
+        ELSEIF (in_auto_config = 'full') THEN
+            UPDATE performance_schema.setup_consumers
+                SET ENABLED = 'YES';
+
+            UPDATE performance_schema.setup_instruments
+                SET ENABLED = 'YES',
+                    TIMED   = 'YES';
+
+            UPDATE performance_schema.threads
```

Comment:
This update, and the one in the medium config are both the same, it can live outside of the IF .. ELSEIF .. just once.